### ISSUE TITLE
Retrying download from history throws database error

### ISSUE THREAD

**Issue by jihbuomj** [evilhero/mylar#1658]
When retrying an issue from history, error is thrown. I believe a portion of the query on line 1256 of webserve.py simply needs to be changed from 

`b.IssueNumber` 
to
`b.Issue_Number`

> Uncaught exception: Traceback (mos                                                                                                                    t recent call last):
>   File "/app/mylar/mylar/logger.py", line 158, in new_run
>     old_run(*args, **kwargs)
>   File "/usr/lib/python2.7/threading.py", line 754, in run
>     self.__target(*self.__args, **self.__kwargs)
>   File "/app/mylar/mylar/webserve.py", line 1256, in retryissue
>     ckthis = myDB.selectone('SELECT a.ComicID, a.ComicName, a.ComicVersion, a.ComicYear, b.IssueID, b.IssueNumber, b.IssueDate FROM comics as a INNER JOIN annuals as b ON a.ComicID = b.ComicID WHERE IssueID=?', [IssueID]).fetchone()
>   File "/app/mylar/mylar/db.py", line 153, in selectone
>     sqlResults = self.fetch(query, args)
>   File "/app/mylar/mylar/db.py", line 94, in fetch
>     sqlResult = cursor.execute(query, args)
> OperationalError: no such column: b.IssueNumber

Similar change might need to be done on line 1258 as the `issues` table also uses an underscore.

**Comment by jihbuomj** [comment]
There were a few more errors that popped up after fixing that on my machine. I was able to successfully retry downloads from history when I changed lines 1256-1269 of webserve.py to:

```
else:
	chkthis = myDB.selectone('SELECT a.ComicID, a.ComicName, a.ComicVersion, a.ComicYear, b.IssueID, b.Issue_Number, b.IssueDate FROM comics as a INNER JOIN annuals as b ON a.ComicID = b.ComicID WHERE IssueID=?', [IssueID]).fetchone()
	if chkthis is None:
		chkthis = myDB.selectone('SELECT a.ComicID, a.ComicName, a.ComicVersion, a.ComicYear, b.IssueID, b.Issue_Number, b.IssueDate FROM comics as a INNER JOIN issues as b ON a.ComicID = b.ComicID WHERE IssueID=?', [IssueID]).fetchone()
		modcomicname = chkthis['ComicName']
	else:
		modcomicname = chkthis['ComicName'] + ' Annual'
	
	comicinfo = []
	comicinfo.append({ "ComicName":     chkthis['ComicName'],
                           "ComicVolume":   chkthis['ComicVersion'],
                           "IssueNumber":   chkthis['Issue_Number'],
                           "comyear":       chkthis['ComicYear'],
                           "IssueDate":     chkthis['IssueDate'],
                           "pack":          False,
                           "modcomicname":  modcomicname})
```

I don't know if this breaks anything else, but hopefully this helps resolve the issue.

**Comment by evilhero** [comment]
Yeah that makes sense - I believe I started to try and fix some of the history items and point to a new method, and either ran out of time or just plain forgot to finish (hey, I'm only human). 

Thanks much for the detailed analysis and fix too! I'll mod my local repo and see if it's all working on this end - I don't think those values would impact anything at that stage, but I've been caught by the Issue_Number / IssueNumber problem before. I'll have to try and fix that at some point as it's easier for consistency obviously especially for those helping out with code fixes who aren't aware of the slight variable change. 

Thanks again! I'll report back if I find any issues just so you're aware, otherwise I'll push a fix to dev probably tomorrow and give you the credit for the fix ;)

**Comment by jihbuomj** [comment]
No problem, really appreciate all the work you've done so far, just started using this downloader and it works really well. My comic book addiction thanks you.

**Comment by teamcoltra** [comment]
👍 also having this issue.

**Comment by hutch0** [comment]
Same here. Just closed my issue which wasn't reproducible. I had the IssueNumber problem, added the db column, and got the chkthis error. I should have searched before posting.

**Comment by evilhero** [comment]
This should all be fixed now in the push that just went up. I tested retry against torrents that were both on a watchlist and were one-off's, as well as tested it against nzb snatches both on a watchlist and as one-off's. I also tested from both history and series pages. 

I did quick testing on each one to make sure it wouldn't throw an error, or if it did log it properly and continue without crashing as well as being successfully retried. Obviously, I might have missed something, but I think I got it working now.

Let me know if something in the retry option isn't working and I can look at it in more detail.
